### PR TITLE
TF 12, Vault 1.2.0, Jenkinsfile formatting

### PR DESCRIPTION
**Pull request by bgeesaman**
Terraform v.12+ syntax, Vault 1.0.0->1.2.0 changes, Jenkinsfile formatting and pod template.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -63,18 +62,18 @@ resource "google_container_cluster" "app" {
 
   node_config {
     machine_type    = "n1-standard-1"
-    service_account = "${google_service_account.app_cluster.email}"
+    service_account = google_service_account.app_cluster.email
 
     oauth_scopes = [
       "https://www.googleapis.com/auth/cloud-platform",
```
Comment (chrislovecnm) [inline terraform/main.tf]: Not related, but is all we need?

**Review by chrislovecnm — COMMENTED** [review]
Just a quick question

File: terraform/main.tf (hunk 1 above)
Comment (bgeesaman) [reply terraform/main.tf]: It's technically more than we need, so I'll amend this commit and work in the default IAM scopes.  Right now, there is a dedicated SA with the `any` oauth scope but no actual IAM permissions assigned to it.  It's technically very secure this way, but the cluster is most likely unable to send logs/metrics, etc.

File: test/make.sh [hunk 2]
```diff
@@ -48,8 +48,10 @@ function docker() {
 # files ending in '.tf'
 function check_terraform() {
   echo "Running terraform validate"
-  #shellcheck disable=SC2156
-  find . -name "*.tf" -exec bash -c 'terraform validate --check-variables=false $(dirname "{}")' \;
+  REPO_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+  cd "${REPO_ROOT}/terraform" || exit
+  terraform init
+  terraform validate
```
Comment (chrislovecnm) [inline test/make.sh:54]: we are gonna need to do this across all the projects, can you ping Kristen and backlog this please?

**Review by chrislovecnm — APPROVED** [review]
/lgtm

**Pull request merged**